bound/typescript/src/crypto/key_managers/index.ts

```diff
@@ -0,0 +1,35 @@
+import { catchWeb5Error } from "../../errors";
+import wasm from "../../wasm";
+import { Signer, TypescriptSigner } from "../dsa";
+import { Jwk } from "../jwk";
+
+export interface KeyManager {
+  importPrivateJwk(privateJwk: Jwk): Jwk
+  getSigner(publicJwk: Jwk): Signer
+}
+
+export class TypescriptKeyManager implements KeyManager {
+  private wasmKeyManager: wasm.WasmKeyManager;
+
+  constructor(wasmKeyManager: wasm.WasmKeyManager) {
+    this.wasmKeyManager = wasmKeyManager
+  }
+
+  importPrivateJwk(privateJwk: Jwk): Jwk {
+    try {
+      const wasmJwk = this.wasmKeyManager.import_private_jwk(privateJwk.toWasmJwk());
+      return Jwk.fromWasmJwk(wasmJwk);
+    } catch (error) {
+      throw catchWeb5Error(error);
```

Comment:
its strange throwing a function, 

Chad says we can do this:
```
// Define the decorator
function CatchWeb5Error(
  target: Object,
  propertyKey: string,
  descriptor: PropertyDescriptor
) {
  const originalMethod = descriptor.value;

  descriptor.value = function (...args: any[]) {
    try {
      return originalMethod.apply(this, args);
    } catch (error) {
      throw catchWeb5Error(error);
    }
  };

  return descriptor;
}

// Apply the decorator to your methods
class YourClass {
  @CatchWeb5Error
  computeThumbprint(): string {
    const wasmJwk = this.toWasmJwk();
    return wasmJwk.compute_thumbprint();
  }

  @CatchWeb5Error
  anotherMethod(): void {
    // Your method implementation
  }
}
```

Reply:
We can think about it and if we want to go this way or another can make an issue to update error handling across the board

Reply:
But yeah, in a follow up PR

Reply:
looks like decorators are still considered an experimental feature so I may settle on a higher order function for now